gym/lib/gym/runner.rb

```diff
@@ -150,11 +150,20 @@ def compress_and_move_dsym
       if Dir.exist?(bcsymbolmaps_directory)
         UI.message("Mapping dSYM(s) using generated BCSymbolMaps") unless Gym.config[:silent]
         available_dsyms.each do |dsym|
-          command = []
-          command << "dsymutil"
-          command << "--symbol-map #{bcsymbolmaps_directory.shellescape}"
-          command << dsym.shellescape
-          Helper.backticks(command.join(" "), print: !Gym.config[:silent])
+          dwarfdump_command = []
+          dwarfdump_command << "dwarfdump"
+          dwarfdump_command << "--uuid #{dsym.shellescape}"
+
+          dwarfdump_result = Helper.backticks(dwarfdump_command.join(" "), print: false)
+          architecture_uuids = dwarfdump_result.split("\n").map { |info| info.split(" ")[1] }
+
+          architecture_uuids.each do |uuid|
+            command = []
+            command << "dsymutil"
+            command << "--symbol-map #{bcsymbolmaps_directory.shellescape}/#{uuid}.bcsymbolmap"
+            command << dsym.shellescape
+            Helper.backticks(command.join(" "), print: !Gym.config[:silent])
+          end
```

Comment:
1. Can we add a comment on why we are splitting on `[1]`? (see below)
2. Can we also do something like below where we map the `architecture_uuids` to an array of paths and then iterate over the paths? This way we can easily default back to original implementation if there are no uuids.

```rb
          dwarfdump_command = []
          dwarfdump_command << "dwarfdump"
          dwarfdump_command << "--uuid #{dsym.shellescape}"

          # Gets UUIDs for all architectures using dwarddump
          dwarfdump_result = Helper.backticks(dwarfdump_command.join(" "), print: false)
          architecture_uuids = dwarfdump_result.split("\n").map { |info| info.split(" ")[1] }

          symbol_map_paths = architecture_uuids.map do |uuid|
            "#{bcsymbolmaps_directory.shellescape}/#{uuid}.bcsymbolmap"
          end

          symbol_map_paths << bcsymbolmaps_directory.shellescape if symbol_map_paths.empty?

          symbol_map_paths.each do |path|
            command = []
            command << "dsymutil"
            command << "--symbol-map #{path}"
            command << dsym.shellescape
            Helper.backticks(command.join(" "), print: !Gym.config[:silent])
          end
```

Thoughts? 😊

Reply:
Thank you for contributing! It is definitely better to add some additional checks.

If we expect `dwarddump` to not return UUIDs in some cases or return an invalid text, we might add a verification for `dwarddump` result. Regex is based on [this](https://developer.apple.com/library/archive/technotes/tn2151/_index.html) technical note:

> Convert the UUID of the binary image to a 32 character string seperated in groups of 8-4-4-4-12 (XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX). Note that all letters must be uppercased.

AppStore dSYM files have lowercased names, so we could be not so strict as well and allow lowercased symbols. As result: 

```ruby
uuid_regex = /[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}/

if Dir.exist?(bcsymbolmaps_directory)
  UI.message("Mapping dSYM(s) using generated BCSymbolMaps")
  available_dsyms.each do |dsym|
    dwarfdump_command = []
    dwarfdump_command << "dwarfdump"
    dwarfdump_command << "--uuid #{dsym.shellescape}"

    dwarfdump_result = Helper.backticks(dwarfdump_command.join(" "), print: false)
    architecture_infos = dwarfdump_result.split("\n")
    architecture_uuids = architecture_infos.map { |info|
      info_array = info.split(" ")
      uuid = info_array[1]

      if uuid.nil? || !uuid.match(uuid_regex)
        nil
      else
        uuid
      end
    }.reject{ |uuid| uuid.nil? }

    symbol_map_paths = architecture_uuids.map do |uuid|
      "#{bcsymbolmaps_directory.shellescape}/#{uuid}.bcsymbolmap"
    end

    symbol_map_paths << bcsymbolmaps_directory.shellescape if symbol_map_paths.empty?

    symbol_map_paths.each do |path|
      command = []
      command << "dsymutil"
      command << "--symbol-map #{path}"
      command << dsym.shellescape
      Helper.backticks(command.join(" "), print: !Gym.config[:silent])
    end
  end
end
```

Reply:
So I've tested and pushed this combined code above with additional `dwarfdump` output validation and a fallback to the original implementation. Also applied `rubocop` corrections. 

I'd avoid adding any comments, just instead to make the variable names more explicit and readable. For example, splitting on `[1]` is `uuid = info_array[1]` now. Please reopen the conversation if any additional changes are required.